### PR TITLE
Bit Querying works

### PR DESCRIPTION
**Pull request by remyschwab**
Perform logical operations on bitvectors instead of counting agent to find bins with hits.

### CODE REVIEW

File: include/utils.h [hunk 1]
```diff
@@ -1,5 +1,4 @@
-#ifndef UTILS_H
-#define UTILS_H
+#pragma once
```
Comment (vmusch) [inline include/utils.h:2]: give it a big difference between include Guards and pragma once?
i found a heated discussion about it  https://stackoverflow.com/questions/1143936/pragma-once-vs-include-guards but i think both are equal. But if you want we can switch all headers to #pragma once.

**Review by vmusch — COMMENTED** [review]
i think the changes looks good and really clean :)

File: include/utils.h (hunk 1 above)
Comment (remyschwab) [reply include/utils.h:2]: Haha this was just me desperately trying to figure out what was wrong with my code... I agree they're both probably equal but yea we should just stick to one across this project. We can do that in another PR though

**Comment by remyschwab** [comment]
> i think the changes looks good and really clean :)

Great! Can you formally approve the PR here then? I set up this repo so that all PRs need to be seen and approved by one of us

**Review by vmusch — APPROVED** [review]

**Pull request merged**